### PR TITLE
Improvements based on partial audit

### PR DESCRIPTION
**Pull request by tomazmm**
## Description
Implemented some improvements that were proposed by spleen and some other modifications, that I think it makes code better looking.

## Improvements

Readability, critical bug fixing.

## How Has This Been Tested?

Testing with the existing test-cases.

## Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: score/liquid_icx/liquid_icx.py [hunk 1]
```diff
@@ -364,12 +347,11 @@ def _join(self, sender: Address, value: int) -> None:
         :param value: Amount of ICX to join the pool
         """
 
-        node_id = None
         wallet = Wallet(self.db, sender)
         if wallet.node_id == 0:
```
Comment (mr-rooftop) [inline score/liquid_icx/liquid_icx.py:351]: also use `exists()` here

**Review by mr-rooftop — APPROVED** [review]
Splendid readability improvements! I like the refactor of node_id
Test cases passed on local.

Mentioned a few things to consider changing.
Merge once you addressed the potential changes

File: score/liquid_icx/liquid_icx.py [hunk 2]
```diff
@@ -112,9 +112,7 @@ def lockedOf(self, _owner: Address) -> int:
         return Wallet(self.db, _owner).locked
 
     @external(readonly=True)
-    def getWallet(self, _address: Address = None) -> dict:
-        if _address is None:
```
[the remark this answers is not in the document]
Comment (tomazmm) [reply score/liquid_icx/liquid_icx.py:116]: No, you can not pass None to the function. it needs to be Address object

File: score/liquid_icx/liquid_icx.py [hunk 3]
```diff
@@ -255,8 +236,10 @@ def claim(self) -> None:
         claim_amount = wallet.claim()
 
         if claim_amount:
-            self.icx.send(self.msg.sender, claim_amount)
-            self.Claim()
+            if self.icx.send(self.msg.sender, claim_amount):
```
[the remark this answers is not in the document]
Comment (tomazmm) [reply score/liquid_icx/liquid_icx.py:239]: True, but then I could not revert with a custom message. But we would definitely skip this and revert with a default one.

**Pull request merged**